lightningd/bitcoind.c

```diff
@@ -783,6 +783,66 @@ void bitcoind_gettxout(struct bitcoind *bitcoind,
 			  NULL);
 }
 
+static bool extract_numeric_version(struct bitcoin_cli *bcli,
+			    const char *output, size_t output_bytes,
+			    u64 *version)
+{
+	const jsmntok_t *tokens, *versiontok;
+	bool valid;
+
+	tokens = json_parse_input(output, output, output_bytes, &valid);
+	if (!tokens)
+		fatal("%s: %s response",
+		      bcli_args(tmpctx, bcli),
+		      valid ? "partial" : "invalid");
+
+	if (tokens[0].type != JSMN_OBJECT) {
+		log_unusual(bcli->bitcoind->log,
+			    "%s: gave non-object (%.*s)?",
```

Comment:
"(%.*s)?"   wrong format?

Reply:
Correct format.  `%.*s` means get two arguments: first is length of string, second is string (which might not be null terminated).

Reply:
Thanks, glad to known new tricks.